src/main/java/frc/robot/subsystem/ElevatorSubsystem.java

```diff
@@ -0,0 +1,366 @@
+
+        MoUtils.setupRelativeEncoder(
+                elevatorRelEncoder,
+                elevatorAbsEncoder.getPosition(),
+                MoPrefs.elevatorAbsZero.get(),
+                MoPrefs.elevatorEncoderScale.get());
+
+        MoUtils.setupRelativeEncoder(
+                wristRelEncoder,
+                wristAbsEncoder.getPosition(),
+                MoPrefs.wristAbsZero.get(),
+                MoPrefs.wristEncoderScale.get());
+
+        elevatorAbsEncoder.setConversionFactor(MoPrefs.elevatorAbsEncoderScale.get());
+
+        elevatorVelTuner.populatePIDValues();
+        wristVelTuner.populatePIDValues();
+        elevatorPosTuner.populatePIDValues();
+        wristPosTuner.populatePIDValues();
+    }
+
+    public ElevatorSubsystem() {
+        super("Elevator");
+
+        this.elevatorA = new SparkMax(Constants.ELEVATORA.address(), MotorType.kBrushless);
+        this.elevatorB = new SparkMax(Constants.ELEVATORB.address(), MotorType.kBrushless);
+        this.elevatorWrist = new SparkMax(Constants.ELEVATOR_WRIST.address(), MotorType.kBrushless);
+        this.endEffector = new VictorSPX(Constants.END_EFFECTOR.address());
+
+        elevatorAbsEncoder = MoRotationEncoder.forSparkAbsolute(elevatorA, Units.Rotations);
+        wristAbsEncoder = MoRotationEncoder.forSparkAbsolute(elevatorWrist, Units.Rotations);
+
+        elevatorRelEncoder = MoDistanceEncoder.forSparkRelative(elevatorA, Units.Centimeters);
+        wristRelEncoder = MoRotationEncoder.forSparkRelative(elevatorWrist, Units.Rotations);
+
+        // Setup listeners for encoder scales and absolute zeros. Use notifyImmediately on zero listeners to set the
+        // values now.
+        MoPrefs.elevatorEncoderScale.subscribe(scale -> MoUtils.setupRelativeEncoder(
+                elevatorRelEncoder, elevatorAbsEncoder.getPosition(), MoPrefs.elevatorAbsZero.get(), scale));
+        MoPrefs.elevatorAbsEncoderScale.subscribe(elevatorAbsEncoder::setConversionFactor, true);
+        MoPrefs.wristEncoderScale.subscribe(scale -> MoUtils.setupRelativeEncoder(
+                wristRelEncoder, wristAbsEncoder.getPosition(), MoPrefs.wristAbsZero.get(), scale));
+        MoPrefs.elevatorAbsZero.subscribe(zero -> MoUtils.setupRelativeEncoder(
+                elevatorRelEncoder, elevatorAbsEncoder.getPosition(), zero, MoPrefs.elevatorEncoderScale.get()));
+        MoPrefs.wristAbsZero.subscribe(zero -> MoUtils.setupRelativeEncoder(
+                wristRelEncoder, wristAbsEncoder.getPosition(), zero, MoPrefs.wristEncoderScale.get()));
+
+        MoPrefs.elevatorMaxExtension.subscribe(limit ->
+                elevatorLimitConfig.forwardSoftLimit((float) limit.in(elevatorRelEncoder.getInternalEncoderUnits())));
+        MoPrefs.wristMaxExtension.subscribe(limit ->
+                wristLimitConfig.forwardSoftLimit((float) limit.in(wristRelEncoder.getInternalEncoderUnits())));
```

Comment:
This won't quite work.

Basically, we want to do 2 things:
1. immediately configure the soft limits on the motor, since we're setting up the motor for the first time
2. reconfigure the soft limits on the motor any time the value changes

Unfortunately, this doesn't do either of those things:
1. we don't specify `notifyImmediately`, so it defaults to false. So we don't set the soft limit on the config until it changes
2. when the value changes, we update the config. But we never re-write that config to the motor controllers, so that value is not populated

I suggest you update `configureMotors()` to also set the forward soft limits on the configurations.
```java
        elevatorLimitConfig.forwardSoftLimit((float) MoPrefs.elevatorMaxExtension.get().in(elevatorRelEncoder.getInternalEncoderUnits()));
        wristLimitConfig.forwardSoftLimit((float) MoPrefs.wristMaxExtension.get().in(wristRelEncoder.getInternalEncoderUnits()));
```

Then you can just call `configureMotors` in the listener.

```suggestion
        MoPrefs.elevatorMaxExtension.subscribe(limit ->  configureMotors());
        MoPrefs.wristMaxExtension.subscribe(limit -> configureMotors());
```